### PR TITLE
Update Pumpkin

### PR DESCRIPTION
**Pull request by good777LUCKY**
Update pumpkin to use the new block state system and add a carve behavior. (Add #662)
Also, rename "Caved Pumpkin" to "Carved Pumpkin".

### CODE REVIEW

**Review by LoboMetalurgico — APPROVED** [review]
Thanks for your Pull Request

**Review by LoboMetalurgico — CHANGES_REQUESTED** [review]
During the test of the game, I noticed that the pumpkins disintegrate as soon as they are activated by the redstone. Could you investigate this, please?

**Comment by good777LUCKY** [comment]
> During the test of the game, I noticed that the pumpkins disintegrate as soon as they are activated by the redstone. Could you investigate this, please?

I fix it.

File: src/main/java/cn/nukkit/block/BlockPumpkin.java [hunk 1]
```diff
@@ -1,20 +1,30 @@
 package cn.nukkit.block;
 
 import cn.nukkit.Player;
+import cn.nukkit.block.BlockCarvedPumpkin;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/block/BlockPumpkin.java:4]: This import is not being used (cn.nukkit.block.BlockCarvedPumpkin). Could you remove it?

File: src/main/java/cn/nukkit/block/BlockPumpkin.java [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
+import cn.nukkit.blockproperty.BlockProperties;
 import cn.nukkit.item.Item;
 import cn.nukkit.item.ItemBlock;
+import cn.nukkit.item.ItemID;
+import cn.nukkit.item.ItemSeedsPumpkin;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/block/BlockPumpkin.java:9]: This import (cn.nukkit.item.ItemSeedsPumpkin) is not being used. Could you remove it?

File: src/main/java/cn/nukkit/block/BlockPumpkin.java [hunk 3]
```diff
@@ -32,15 +42,20 @@ public String getName() {
     public int getId() {
         return PUMPKIN;
     }
-
+    
+    @Override
+    public BlockProperties getProperties() {
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/block/BlockPumpkin.java:47]: This function requires the @Nonnull annotation

**Review by LoboMetalurgico — DISMISSED** [review]
Thanks for the correction. I found some more simple things that need to be fixed before we merge.

**Review by LoboMetalurgico — CHANGES_REQUESTED** [review]
Thanks for the correction. I found some more simple things that need to be fixed before we merge.

**Comment by good777LUCKY** [comment]
> Thanks for the correction. I found some more simple things that need to be fixed before we merge.

Fixed

**Review by LoboMetalurgico — APPROVED** [review]

**Comment by LoboMetalurgico** [comment]
Thank You!

**Pull request merged**